### PR TITLE
enhancement: max active runs and and catchup toggle to schedule UI

### PR DESCRIPTION
**Pull request by WillRaphaelson**
Adds catchup and max scheduled runs to the schedule edit / create modal.

image

### CODE REVIEW

**Comment by WillRaphaelson** [comment]
closes [#13205](about:blank)

**Comment by abrookins** [comment]
How do you clear max active runs / return to the default?

File: src/components/ScheduleFormModal.vue [hunk 1]
```diff
@@ -23,6 +23,26 @@
...
     <FlowRunJobVariableOverridesLabeledInput v-if="can.access.deploymentScheduleFlowRunInfraOverrides" v-model="internalJobVariables" />
 
+    <p-label
+      label="Max active runs"
+      :message="maxActiveRunsErrorMessage"
+      :state="maxActiveRunsState"
+      class="interval-schedule-form__column--span-3"
+    >
+      <p-number-input v-model="internalMaxActive" min="0" :state="maxActiveRunsState" />
+    </p-label>
```
Comment (znicholasbrown) [inline src/components/ScheduleFormModal.vue:33]: I don't think there's a way to nullify this value rn - any schedule created will have a value for this, defaulting to 0. I'd recommend we take page from the schemas work and provide a way to omit this value.

File: src/components/DeploymentScheduleCard.vue [hunk 2]
```diff
@@ -28,13 +28,14 @@
   const props = defineProps<{
     deployment: Deployment,
     deploymentSchedule: DeploymentSchedule,
+    deploymentScheduleCompatible: DeploymentScheduleCompatible,
```
Comment (znicholasbrown) [inline src/components/DeploymentScheduleCard.vue:31]: I'm not sure I understand what this prop is doing - the `DeploymentSchedule` has the `catchup` and `maxActiveRuns` values, right? Is there a reason we can't pull them from that prop? I may be misunderstanding how these are used.

**Review by znicholasbrown — COMMENTED** [review]
A couple of questions @WillRaphaelson, this looks p good!

**Comment by abrookins** [comment]
The catchup field probably needs a better label, even with the tool tip. It's a little inscrutable. Maybe "Catch up late runs?"

**Pull request closed without merging**